src/main/java/duke/parts/Storage.java

```diff
@@ -0,0 +1,100 @@
+package duke.parts;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Used to help store, save and load user data.
+ * This allows data to be preserved when the system is turned off.
+ */
+public class Storage {
+    String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Checks if the filePath is valid and creates a directory
+     * and file when it is not present.
+     * @throws IOException
+     */
+    void createFile() throws IOException {
```

Comment:
Is `IOException` really thrown? I think you've dealt with it using `catch` right?